src/Dotnet.Installer.Console/Commands/ListCommand.cs

```diff
@@ -49,57 +63,115 @@ private async Task Handle(bool includeUnsupported)
                     eolString);
-
+                
                 continue;
-
-                string ComponentStatus(string key, int majorVersion, bool isEol)
+                
+                async Task<string> ComponentStatus(string name, int majorVersion)
                 {
-                    if (components.TryGetValue(key, out var component))
+                    var component = majorVersionGroup
+                        .FirstOrDefault(c => c.Name == name && c.MajorVersion == majorVersion);
+                    
+                    bool isInstalled = component is { Installation: not null };
+
+                    string status;
+                    
+                    if (component is null)
                     {
-                        return $"[bold green]Installed [[{component.Version}]][/]";
+                        status = "[grey]-";
                     }
+                    else if (isInstalled)
+                    {
+                        status = "[green][bold]Installed[/]";
+                    }
+                    else // is available for installation
+                    {
+                        status = "[blue][bold]Available[/]";
+                    }
+                    
+                    DotnetVersion? version = await GetComponentVersion(component).ConfigureAwait(false);
 
-                    var available = _manifestService.Remote.Any(c => c.Name == key && c.MajorVersion == majorVersion);
+                    status += version is null ? "[/]" : $" [[{version}]][/]";
+                    return status;
+                }
 
-                    if (!available)
+                async Task<DotnetVersion?> GetComponentVersion(Component? component)
+                {
+                    if (component is null)
                     {
-                        return "[grey]-[/]";
+                        return null;
+                    }
+                    
+                    if (component.Installation is not null)
+                    {
+                        return component.GetLocalDotnetVersion(_manifestService, _fileService);
                     }
 
-                    return isEol ? "[grey]Available[/]" : "[bold blue]Available[/]";
+                    try
+                    {
+                        if (timeoutInMilliseconds == 0)
+                        {
+                            return await component.GetRemoteDotnetVersion(_snapService).ConfigureAwait(false);
+                        }
+                        
+                        using var cancellationTokenSource = new CancellationTokenSource();
+                        var queryVersionTask = component.GetRemoteDotnetVersion(
+                            _snapService, 
+                            cancellationTokenSource.Token);
+                        var timeoutTask = Task.Delay(
+                             TimeSpan.FromMilliseconds(timeoutInMilliseconds), 
+                             cancellationTokenSource.Token);
```

Comment:
**issue (blocking):** Timeouts are being hit very oftenly.

I believe the timeout should be a bit more generous. 800 milliseconds is currently giving very intermittent results as some API requests are taking longer than that for me. As a result, I'm more often than not seeing errors like these:
```
/home/mateus/Git/dotnet-snap/src/Dotnet.Installer.Console/bin/Debug/net8.0/Dotnet.Installer.Console list --all
[15:33:27 ERR] Querying the version for aspnetcore-runtime-60 timed out
┌────────────┬────────────────────┬──────────────────────┬─────────────────────┬─────────────┐
│ Version    │ .NET Runtime       │ ASP.NET Core Runtime │ SDK                 │ End of Life │
├────────────┼────────────────────┼──────────────────────┼─────────────────────┼─────────────┤
│ .NET 6 LTS │ Installed [6.0.36] │ Available            │ Available [6.0.136] │ 11/12/2024  │
│ .NET 8 LTS │ Available [8.0.11] │ Available [8.0.11]   │ Available [8.0.111] │ 11/10/2026  │
│ .NET 9     │ Available [9.0.0]  │ Available [9.0.0]    │ Available [9.0.100] │ 5/12/2026   │
└────────────┴────────────────────┴──────────────────────┴─────────────────────┴─────────────┘
```

**suggestion:** Let's maybe extend the timeout to something between 3 and 5 seconds.

Reply:
I extended the timeout to 5 seconds, although it is probably not needed anymore as I also parallelized the calls to the snap & file service, which significantly improved the query time.